nilearn/connectivity/connectivity_matrices.py

```diff
@@ -0,0 +1,349 @@
+import warnings
+from math import sqrt
+
+import numpy as np
+from scipy import linalg
+
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.covariance import EmpiricalCovariance
+from .._utils.extmath import is_spd
+
+
+def _check_square(matrix):
+    """Raise a ValueError if the input matrix is square.
+
+    Parameters
+    ----------
+    matrix : numpy.ndarray
+        Input array.
+    """
+    if matrix.ndim != 2 or (matrix.shape[0] != matrix.shape[-1]):
+        raise ValueError('Expected a square matrix, got array of shape'
+                         ' {0}.'.format(matrix.shape))
+
+
+def _check_spd(matrix):
```

Comment:
I would add a `raise_error=True` parameter to `is_spd` instead of declaring a whole new function?